lib/rules/validate-quote-marks.js

```diff
@@ -106,10 +106,16 @@ module.exports.prototype = {
                     return;
                 }
 
+                token.value = token.value.replace(
+                    new RegExp('^' + mark + '(.*)' + mark + '$'),
```

Comment:
I'm accepting Pull Requests. ;) I didn't write the logic on this one, I pulled it from another PR and updated #1201 and #1202 here.